docs/02-app/01-building-your-application/07-configuring/05-mdx.mdx

```diff
@@ -127,100 +204,225 @@ Checkout my React component:
 <MyComponent />
 ```
 
-Navigating to the `/my-mdx-page` route should display your rendered MDX.
+Import the MDX file inside the page to display the content:
 
-## Remote MDX
+<AppOnly>
 
-If your markdown or MDX files or content lives _somewhere else_, you can fetch it dynamically on the server. This is useful for content stored in a separate local folder, CMS, database, or anywhere else. A popular community package for this use is [`next-mdx-remote`](https://github.com/hashicorp/next-mdx-remote#react-server-components-rsc--nextjs-app-directory-support).
+```tsx filename="app/mdx-page/page.tsx" switcher
+import Welcome from '@/markdown/welcome.mdx'
 
-> **Good to know**: Please proceed with caution. MDX compiles to JavaScript and is executed on the server. You should only fetch MDX content from a trusted source, otherwise this can lead to remote code execution (RCE).
+export default function Page() {
+  return <Welcome />
+}
+```
 
-The following example uses `next-mdx-remote`:
+```jsx filename="app/mdx-page/page.js" switcher
+import Welcome from '@/markdown/welcome.mdx'
+
+export default function Page() {
+  return <Welcome />
+}
+```
+
+</AppOnly>
+
+<PagesOnly>
+
+```tsx filename="pages/mdx-page.tsx" switcher
+import Welcome from '@/markdown/welcome.mdx'
+
+export default function Page() {
+  return <Welcome />
+}
+```
+
+```jsx filename="pages/mdx-page.js" switcher
+import Welcome from '@/markdown/welcome.mdx'
+
+export default function Page() {
+  return <Welcome />
+}
+```
+
+</PagesOnly>
+
+Navigating to the `/mdx-page` route should display your rendered MDX page.
+
+## Using custom styles and components
+
+One of the benefits of using markdown is that it maps to native HTML elements, making writing fast without having to write code.
```

Comment:
This reads weird to me.

Reply:
Addressed in https://github.com/vercel/next.js/pull/63568/commits/daa0388a313d016a6b74cff02a87550cc5b14bf2.